### PR TITLE
Add chat completions example for LLM API response parsing

### PR DESCRIPTION
**Pull request by joshbeckman**
## Summary
- Added a new Run Code example that parses responses from LLM chat completion APIs
- Supports OpenAI, Anthropic Claude, and Google Gemini APIs
- Provides robust error handling and comprehensive test coverage

## What this PR does
This PR adds a new example to the Flow Code Examples repository that demonstrates how to parse responses from Large Language Model (LLM) APIs when using Shopify Flow's "Send HTTP Request" action.

### Features
- ✅ Parses OpenAI-compatible responses (OpenAI GPT, Google Gemini)
- ✅ Parses Anthropic Claude responses
- ✅ Extracts content, model info, usage statistics, and finish reasons
- ✅ Comprehensive error handling for invalid JSON, API errors, and unknown formats
- ✅ 11 test cases covering all major scenarios
- ✅ Detailed README with provider documentation links and cURL examples

## Testing
All tests pass successfully:
- 11 test cases covering OpenAI, Anthropic, and error scenarios
- Tests handle edge cases like missing usage data and unknown response formats
- Ran full test suite to ensure no regressions

## Documentation
The README includes:
- Links to official API documentation for each provider
- Ready-to-use cURL examples that can be copied into Flow's Send HTTP Request step
- Security best practices for API key management
- Common e-commerce use cases
- Troubleshooting guide

## Related Issue
Closes [Add chat completions example to flow-code-examples](https://github.com/shop/issues-automate/issues/608)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: run-code-examples/http-chat-completions/README.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+The parser handles various error scenarios:
+- Invalid JSON responses
+- Missing required fields
+- API error responses
+- Network failures
+- Unknown response formats
+
+## Security Best Practices
+
+1. **Never hardcode API keys** - Use Flow's environment variables
```
Comment (MaxLeduc) [inline run-code-examples/http-chat-completions/README.md]: Use Flow secrets?

**Review by MaxLeduc — APPROVED** [review]

File: run-code-examples/http-chat-completions/README.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+## How to Use
+
+### Step 1: Configure Send HTTP Request
+
+In your Shopify Flow workflow:
+
+1. Add a "Send HTTP Request" action before the "Run Code" action
+2. Configure it with one of the example cURL requests below
+3. Store your API key in Flow's environment variables for security
```
Comment (joshbeckman) [inline run-code-examples/http-chat-completions/README.md]:
```suggestion
3. Store your API key in Flow's secret storage for security
```

File: run-code-examples/http-chat-completions/README.md (hunk 1 above)
Comment (joshbeckman) [reply run-code-examples/http-chat-completions/README.md]:
```suggestion
1. **Never hardcode API keys** - Use Flow's secret storage
```

**Review by jenniferwills — APPROVED** [review]
